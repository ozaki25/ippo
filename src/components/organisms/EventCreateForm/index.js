import React from 'react';
import { Button, TextField, Typography } from '@material-ui/core';
import propTypes from 'prop-types';

class EventCreateForm extends React.Component {
  state = {
    title: '',
    catchMessage: '',
    place: '',
    hashtag: '',
    startedAt: '',
    endedAt: '',
    loading: false,
    error: false,
  };

  onClick = async event => {
    event.preventDefault();
    const { title, catchMessage, place, hashtag, startedAt, endedAt } = this.state;
    this.setState({ loading: true });
    try {
      const { data } = await this.props.onSubmit({
        title,
        catchMessage,
        place,
        hashtag,
        startedAt,
        endedAt,
      });
      console.log({ data });
    } catch (e) {
      this.setState({ loading: false, error: e.toString() });
    }
  };

  onChange = event => this.setState({ [event.target.name]: event.target.value });

  render() {
    const { title, catchMessage, place, hashtag, startedAt, endedAt, loading, error } = this.state;
    const invalid = this.state.title.trim() === '' || this.state.hashtag.trim() === '';
    return (
      <form>
        {error && <Typography color="error">{error}</Typography>}
        <TextField
          label="イベント名"
          name="title"
          value={title}
          onChange={this.onChange}
          margin="normal"
          color="primary"
          fullWidth
          required
        />
        <br />
        <TextField
          label="概要"
          name="catchMessage"
          value={catchMessage}
          onChange={this.onChange}
          margin="normal"
          color="primary"
          fullWidth
        />
        <br />
        <TextField
          label="場所"
          name="place"
          value={place}
          onChange={this.onChange}
          margin="normal"
          color="primary"
          fullWidth
        />
        <br />
        <TextField
          label="ハッシュタグ"
          name="hashtag"
          value={hashtag}
          onChange={this.onChange}
          margin="normal"
          color="primary"
          fullWidth
          required
        />
        <br />
        <TextField
          label="開始日時"
          name="startedAt"
          type="datetime-local"
          value={startedAt}
          onChange={this.onChange}
          InputLabelProps={{
            shrink: true,
          }}
          margin="normal"
          color="primary"
          fullWidth
        />
        <br />
        <TextField
          label="終了日時"
          name="endedAt"
          type="datetime-local"
          value={endedAt}
          onChange={this.onChange}
          InputLabelProps={{
            shrink: true,
          }}
          margin="normal"
          color="primary"
          fullWidth
        />
        <Button
          onClick={this.onClick}
          disabled={invalid || loading}
          color="primary"
          variant="contained"
          fullWidth
        >
          作成
        </Button>
      </form>
    );
  }
}

EventCreateForm.displayName = 'EventCreateForm';

EventCreateForm.propTypes = {
  onSubmit: propTypes.func.isRequired,
};

EventCreateForm.defaultProps = {};

export default EventCreateForm;
