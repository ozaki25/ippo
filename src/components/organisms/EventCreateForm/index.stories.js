import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import EventCreateForm from '.';

const stories = storiesOf('organisms/EventCreateForm', module);

stories.add('通常パターン場合', () => <EventCreateForm onSubmit={action('onSubmit')} />);

stories.add('エラー', () => (
  <EventCreateForm
    onSubmit={() =>
      new Promise((resolve, reject) => reject(new Error('TypeError: Failed to fetch')))
    }
  />
));
