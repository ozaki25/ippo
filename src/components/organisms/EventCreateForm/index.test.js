import React from 'react';
import { snapshot } from 'src/test/helpers';
import EventCreateForm from '.';

const props = {
  onSubmit: jest.fn(),
};

snapshot('EventCreateForm/nomal', <EventCreateForm {...props} />);
