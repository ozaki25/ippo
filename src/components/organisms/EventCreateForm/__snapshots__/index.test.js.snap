// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: EventCreateForm/nomal 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <EventCreateForm
    onSubmit={[MockFunction]}
  >
    <form>
      <Blueprint3.FormGroup
        label="イベント名"
        labelFor="title"
        labelInfo="(必須)"
      >
        <div
          className="bp3-form-group"
        >
          <label
            className="bp3-label"
            htmlFor="title"
          >
            イベント名
             
            <span
              className="bp3-text-muted"
            >
              (必須)
            </span>
          </label>
          <div
            className="bp3-form-content"
          >
            <Blueprint3.InputGroup
              id="title"
              large={true}
              name="title"
              onChange={[Function]}
              value=""
            >
              <div
                className="bp3-input-group bp3-large"
              >
                <Blueprint3.Icon />
                <input
                  className="bp3-input"
                  id="title"
                  name="title"
                  onChange={[Function]}
                  style={
                    Object {
                      "paddingRight": 10,
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </Blueprint3.InputGroup>
          </div>
        </div>
      </Blueprint3.FormGroup>
      <Blueprint3.FormGroup
        label="概要"
        labelFor="catchMessage"
      >
        <div
          className="bp3-form-group"
        >
          <label
            className="bp3-label"
            htmlFor="catchMessage"
          >
            概要
             
            <span
              className="bp3-text-muted"
            />
          </label>
          <div
            className="bp3-form-content"
          >
            <Blueprint3.InputGroup
              id="catchMessage"
              large={true}
              name="catchMessage"
              onChange={[Function]}
              value=""
            >
              <div
                className="bp3-input-group bp3-large"
              >
                <Blueprint3.Icon />
                <input
                  className="bp3-input"
                  id="catchMessage"
                  name="catchMessage"
                  onChange={[Function]}
                  style={
                    Object {
                      "paddingRight": 10,
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </Blueprint3.InputGroup>
          </div>
        </div>
      </Blueprint3.FormGroup>
      <Blueprint3.FormGroup
        label="場所"
        labelFor="place"
      >
        <div
          className="bp3-form-group"
        >
          <label
            className="bp3-label"
            htmlFor="place"
          >
            場所
             
            <span
              className="bp3-text-muted"
            />
          </label>
          <div
            className="bp3-form-content"
          >
            <Blueprint3.InputGroup
              id="place"
              large={true}
              name="place"
              onChange={[Function]}
              value=""
            >
              <div
                className="bp3-input-group bp3-large"
              >
                <Blueprint3.Icon />
                <input
                  className="bp3-input"
                  id="place"
                  name="place"
                  onChange={[Function]}
                  style={
                    Object {
                      "paddingRight": 10,
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </Blueprint3.InputGroup>
          </div>
        </div>
      </Blueprint3.FormGroup>
      <Blueprint3.FormGroup
        label="ハッシュタグ"
        labelFor="hashtag"
        labelInfo="(必須)"
      >
        <div
          className="bp3-form-group"
        >
          <label
            className="bp3-label"
            htmlFor="hashtag"
          >
            ハッシュタグ
             
            <span
              className="bp3-text-muted"
            >
              (必須)
            </span>
          </label>
          <div
            className="bp3-form-content"
          >
            <Blueprint3.InputGroup
              id="hashtag"
              large={true}
              name="hashtag"
              onChange={[Function]}
              value=""
            >
              <div
                className="bp3-input-group bp3-large"
              >
                <Blueprint3.Icon />
                <input
                  className="bp3-input"
                  id="hashtag"
                  name="hashtag"
                  onChange={[Function]}
                  style={
                    Object {
                      "paddingRight": 10,
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </Blueprint3.InputGroup>
          </div>
        </div>
      </Blueprint3.FormGroup>
      <Blueprint3.FormGroup
        label="開始日時"
        labelFor="startedAt"
      >
        <div
          className="bp3-form-group"
        >
          <label
            className="bp3-label"
            htmlFor="startedAt"
          >
            開始日時
             
            <span
              className="bp3-text-muted"
            />
          </label>
          <div
            className="bp3-form-content"
          >
            <Blueprint3.InputGroup
              id="startedAt"
              large={true}
              name="startedAt"
              onChange={[Function]}
              placeholder="2018/1/23 9:30"
              value=""
            >
              <div
                className="bp3-input-group bp3-large"
              >
                <Blueprint3.Icon />
                <input
                  className="bp3-input"
                  id="startedAt"
                  name="startedAt"
                  onChange={[Function]}
                  placeholder="2018/1/23 9:30"
                  style={
                    Object {
                      "paddingRight": 10,
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </Blueprint3.InputGroup>
          </div>
        </div>
      </Blueprint3.FormGroup>
      <Blueprint3.FormGroup
        label="終了日時"
        labelFor="endedAt"
      >
        <div
          className="bp3-form-group"
        >
          <label
            className="bp3-label"
            htmlFor="endedAt"
          >
            終了日時
             
            <span
              className="bp3-text-muted"
            />
          </label>
          <div
            className="bp3-form-content"
          >
            <Blueprint3.InputGroup
              id="endedAt"
              large={true}
              name="endedAt"
              onChange={[Function]}
              placeholder="2018/1/23 9:30"
              value=""
            >
              <div
                className="bp3-input-group bp3-large"
              >
                <Blueprint3.Icon />
                <input
                  className="bp3-input"
                  id="endedAt"
                  name="endedAt"
                  onChange={[Function]}
                  placeholder="2018/1/23 9:30"
                  style={
                    Object {
                      "paddingRight": 10,
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </Blueprint3.InputGroup>
          </div>
        </div>
      </Blueprint3.FormGroup>
      <Blueprint3.Button
        disabled={true}
        fill={true}
        large={true}
        onClick={[Function]}
        text="作成"
      >
        <button
          className="bp3-button bp3-disabled bp3-fill bp3-large"
          disabled={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          tabIndex={-1}
          type="button"
        >
          <Blueprint3.Icon
            key="leftIcon"
          />
          <span
            className="bp3-button-text"
            key="text"
          >
            作成
          </span>
          <Blueprint3.Icon
            key="rightIcon"
          />
        </button>
      </Blueprint3.Button>
    </form>
  </EventCreateForm>
</MockRouter>
`;
